File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.js
function solution(arr, divisor) {
    const list = arr.reduce((acc, curr) => {
        if(curr % divisor === 0) {
            acc.push(curr);
        }
    return acc;
    }, []);

    if(list.length === 0) {
        return [-1];
    }

    console.log(list);
    list.sort((a,b) => a - b);
    return list;
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

public class Solution {
    public static int[] solution(int[] arr, int divisor) {

        ArrayList<Integer> list = new ArrayList<>();
        for (int i = 0; i < arr.length; i++) {

            if (arr[i] % divisor == 0) {
                list.add(arr[i]);
            }
        }

        if(list.isEmpty()) {
            return new int[]{-1};
        }

        Collections.sort(list);
        int[] result = list.stream().mapToInt(Integer::intValue).toArray();
        System.out.println(Arrays.toString(result));
        return result;
    }

}